src/client.rs

```diff
@@ -187,23 +190,17 @@ impl Client {
             if let Ok(info) = Client::get_node_info(node.url.as_ref(), None).await {
                 if info.status.is_healthy {
                     match network_nodes.get_mut(&info.protocol.network_name) {
-                        Some(network_id_entry) => {
-                            network_id_entry.push((info, node.clone()));
+                        Some(network_node_entry) => {
+                            network_node_entry.push((info, node.clone()));
                         }
-                        None => match &network_info
-                            .read()
-                            .map_or(NetworkInfo::default().network, |info| info.network.clone())
-                        {
-                            Some(id) => {
-                                if info.protocol.network_name.contains(id) {
-                                    network_nodes
-                                        .insert(info.protocol.network_name.clone(), vec![(info, node.clone())]);
-                                }
-                            }
-                            None => {
+                        None => {
+                            let guard = network_info.read().map_err(|_| crate::Error::PoisonError)?;
+                            let id = guard.protocol_parameters.network_name();
+                            // TODO there used to be a some/none match, is this correct?
+                            if info.protocol.network_name.contains(id) {
                                 network_nodes.insert(info.protocol.network_name.clone(), vec![(info, node.clone())]);
```

Comment:
This looks like it will only work if the network_name matches the default we set in the network info?

Reply:
I think it should just `network_nodes.insert(` without the `if info.protocol.network_name.contains(id) {`